packages/framework/src/Component/EntityExtension/EntityExtensionSubscriber.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Component\EntityExtension;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\Event\LoadClassMetadataEventArgs;
+use Doctrine\ORM\Events;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use LogicException;
+use ReflectionClass;
+use Webmozart\Assert\Assert;
+
+/**
+ * Inspired by joschi127/doctrine-entity-override-bundle (https://github.com/joschi127/doctrine-entity-override-bundle)
+ */
+class EntityExtensionSubscriber implements EventSubscriber
+{
+    /**
+     * @var array<class-string, class-string>
+     */
+    protected array $entityExtensionMap;
+
+    /**
+     * @var bool
+     */
+    protected bool $isEntityMapLoaded = false;
+
+    /**
+     * @var array<class-string, array<class-string, class-string>>
+     */
+    protected array $parentClassesByClass = [];
```

Comment:
while I am reviewing the code, I am thinking - is it even necessary to keep an array of all the parents? Would not it be sufficient the keep just these parents that are registered entities? Let's discuss it, please.